src/theory/uf/ho_extension.cpp

```diff
@@ -546,12 +562,27 @@ unsigned HoExtension::checkLazyLambda()
         std::vector<Node> args(flam[0].begin(), flam[0].end());
         Node rhs = d_ll.betaReduce(glam, args);
         Node univ = nm->mkNode(Kind::FORALL, flam[0], lhs.eqNode(rhs));
+        // do quantifier elimination if the option is set
+        if (options().uf.ufHoLambdaQe)
+        {
+          Trace("uf-lambda-qe") << "Given " << flam << " == " << glam << std::endl;
```

Comment:
Can you please add the the description of the PR as documentation here? The description is surprisingly good.